internal/mode/static/state/graph/policy_ancestor.go

```diff
@@ -25,6 +29,26 @@ func ancestorsFull(
 	return true
 }
 
+// ngfPolicyAncestorsFull returns whether or not an ancestor list is full. A list is full when
+// the sum of the following is greater than or equal to the maximum allowed:
+//   - number of non-NGF managed ancestors
+//   - number of NGF managed ancestors already added to the updated list
+//
+// We aren't considering the number of NGF managed ancestors in the current list because the updated list
+// is the new source of truth.
+func ngfPolicyAncestorsFull(policy *Policy, ctlrName string) bool {
```

Comment:
would it be a lot of work to implement partial allowance of NGF ancestors? (like first 3)
just to cover the case when let's say a user add on more to the list  (making it > maxAncetors), at least first 3 will keep working.

Reply:
I think that's how it works with this logic. We loop through each targetRef and attempt to attach it to the target. Once we reach the max ancestors, we return without attaching to the target that we're currently processing. But the previous attachments are still valid.

Reply:
got it. thx!